Packs/ApiModules/Scripts/EWSApiModule/EWSApiModule.py

```diff
@@ -715,3 +876,439 @@ def get_on_prem_version(version: str):
     :return: A Version object representing the on-premises Exchange Server version
     """
     return Version(get_on_prem_build(version))
+
+# Command functions and helpers
+
+
+def is_empty_object(obj) -> bool:
+    """
+    Checks if the given object is empty.
+    :param obj: The object to check.
+
+    :return: True if the object is empty, False otherwise.
+    """
+    return (obj.__sizeof__() if isinstance(obj, map) else len(obj)) == 0
+
+
+def filter_dict_null(d):
+    """
+    Filters out any none values from a dictionary.
+    :param d: The dictionary to filter.
+
+    :return: A new dictionary with all the none values removed.
+    """
+    if isinstance(d, dict):
+        return {k: v for k, v in list(d.items()) if v is not None}
+    return d
+
+
+def switch_hr_headers(obj, hr_header_changes: dict):
+    """
+    Will swap keys according to hr_header_changes.
+    hr_header_changes: a dict, keys are the old value, value is the new value
+    """
+    if not isinstance(obj, dict):
+        return obj
+    obj_copy = obj.copy()
+    for old_header, new_header in hr_header_changes.items():
+        if old_header in obj:
+            obj_copy[new_header] = obj_copy.pop(old_header)
+    return obj_copy
+
+
+def get_entry_for_object(title: str, context_key: str, obj, headers: Optional[list] = None,
+                         hr_header_changes: dict = {}) -> CommandResults:
+    """
+    Create an entry for a given object
+    :param title: Title of the human readable
+    :param context_key: Context key used for entry context
+    :param obj: Object to create entry for
+    :param headers: (Optional) headers used in the tableToMarkDown
+    :param hr_header_changes: (Optional) a dict to map the header names used in the human readable output
+    :return: Entry object to be used with demisto.results()
+    """
+    if is_empty_object(obj):
+        return CommandResults(readable_output='There is no output results')
+    obj = filter_dict_null(obj)
+    hr_obj = switch_hr_headers(obj, hr_header_changes)
```

Comment:
I think resolve_name_command is the only place using hr_header_changes. Maybe its worth moving it to the relevant function?